gen/src/value.rs

```diff
@@ -36,6 +36,16 @@ macro_rules! generate_enum {
 	}
 
 	impl std::fmt::Display for $id {
+	    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		match self {
+		    $(
+			Self::$variant(inner) => write!(f, "{:?}", inner),
```

Comment:
This is about the only line I am not happy with personally. Would have liked to use "{}" instead :disappointed:

Reply:
I agree, it makes the code a bit more confusing as well because the `Display` implementation prints the inner variant’s `Debug`, whereas the `Debug` implementation prints the qualified variant `format!({}({:?}), variant_name, variant_value)`. So this work around works well for printing `Number` but it won’t work for `synth_gen::Value`.

The reason why `{}` can’t be used here is solely because the macro is used on the `Value` enum which has a `Vec` variant which is not display. We could easily refactor these macros later so that case does not happen.